sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/JsonSerializer.java

```diff
@@ -38,20 +38,17 @@
      *
      * @param stream {@link OutputStream} where the JSON tree will be written.
      * @param jsonNode The JSON tree.
-     * @param <S> Type of the output stream.
-     * @return The stream where the JSON tree was written.
      */
-    <S extends OutputStream> S serializeTree(S stream, JsonNode jsonNode);
+    void serializeTree(OutputStream stream, JsonNode jsonNode);
 
     /**
      * Writes a JSON tree into a stream.
      *
      * @param stream {@link OutputStream} where the JSON tree will be written.
      * @param jsonNode The JSON tree.
-     * @param <S> Type of the output stream.
-     * @return The stream where the JSON tree was written.
+     * @return Reactive stream that will indicate operation completion.
      */
-    <S extends OutputStream> Mono<S> serializeTreeAsync(S stream, JsonNode jsonNode);
+    Mono<Void> serializeTreeAsync(OutputStream stream, JsonNode jsonNode);
```

Comment:
I was wondering if the comment in arch board was applicable for async and sync or just sync? Since the point you make about async is still valid when chaining async operators to have the reference to OutputStream.

Reply:
I think it is better to have sync and async return equivalent return types. Having `serialize` being `void` is fine as the `OutputStream` can be passed or newed before calling the API and then returned later.